### PR TITLE
Fix issue #252: build error on AppleClang10 when C++ exceptions are turned off

### PR DESCRIPTION
**Pull request by petamas**
Fix issue #252

### CODE REVIEW

File: include/gsl/gsl-lite.hpp [hunk 1]
```diff
@@ -293,11 +293,28 @@
...
+#   define gsl_HAVE_EXCEPTIONS  1
+#  else
+#   define gsl_HAVE_EXCEPTIONS  0
+#  endif // __EXCEPTIONS && __has_feature(cxx_exceptions)
 # else
-#  define gsl_HAVE_EXCEPTIONS  0
-# endif // __EXCEPTIONS && __has_feature(cxx_exceptions)
+   // clang-cl doesn't define __EXCEPTIONS for MSVC compatibility (see https://reviews.llvm.org/D4065)
+   // Neither does clang in MS-compatiblity mode
+   // Let's hope no one tries to build Objective-C++ code using MS-compatibility mode or clang-cl
```
Comment (mbeutel) [inline include/gsl/gsl-lite.hpp:311]: I haven't heard hear from [Project Islandwood](https://github.com/Microsoft/WinObjC/releases) in quite a while so I guess we shouldn't worry :)

Comment (petamas) [reply include/gsl/gsl-lite.hpp:311]: I should've known there is something like that out there. :D We'll cross that bridge when we get there. I could come up with something that only fails if you use a pre-3.5 clang-cl building ObjectiveC++ with ObjC exceptions enabled, but C++ exceptions disabled, but I don't think it's worth it.

Comment (mbeutel) [reply include/gsl/gsl-lite.hpp:311]: Neither do I.

**Comment by mbeutel** [comment]
Darn, hit Return too fast and messed up the commit message.

**Pull request merged**